CMakeTests.cmake

```diff
@@ -26,7 +26,7 @@
   option (HDF5_DISABLE_TESTS_REGEX "Regex pattern to set execution of specific tests to DISABLED" "")
   mark_as_advanced (HDF5_DISABLE_TESTS_REGEX)
 
-  option (HDF5_TEST_API "Execute HDF5 API tests" OFF)
+  option (HDF5_TEST_API "Execute HDF5 API tests" ON)
```

Comment:
Not sure if this change is worth a release note or not

Reply:
Since it's changing a default, a simple one line should be enough. Definitely not newsletter worthy.